app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java

```diff
@@ -596,12 +599,42 @@ private void setupPhoneUI() {
                 mDelegate.getWindows().getFocusedWindow().loadHome();
             });
 
-            mBinding.headlockToggleButton.setChecked(
-                    SettingsStore.getInstance(PlatformActivity.this).isHeadLockEnabled());
+            SettingsStore settings = SettingsStore.getInstance(PlatformActivity.this);
+            mBinding.headlockToggleButton.setChecked(settings.isHeadLockEnabled());
             mBinding.headlockToggleButton.setOnClickListener(v -> {
                 mDelegate.setLockMode(mBinding.headlockToggleButton.isChecked() ? WidgetManagerDelegate.HEAD_LOCK : WidgetManagerDelegate.NO_LOCK);
             });
 
+            mBinding.windowDistanceButton.setOnClickListener(v -> {
+                LayoutInflater inflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
+                View popupView = inflater.inflate(R.layout.window_distance_popup_layout, null);
+
+                int width = LinearLayout.LayoutParams.WRAP_CONTENT;
+                int height = LinearLayout.LayoutParams.WRAP_CONTENT;
+                boolean focusable = true; // Let taps outside the popup also dismiss it
+                final PopupWindow popupWindow = new PopupWindow(popupView, width, height, focusable);
+
+                SeekBar seekBar = popupView.findViewById(R.id.windowDistancePopupSeekBar);
+                float max = seekBar.getMax();
+                seekBar.setProgress((int) (settings.getWindowDistance() * max));
+                seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+                    @Override
+                    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                        settings.setWindowDistance((float) progress / max);
+                    }
+
+                    @Override
+                    public void onStartTrackingTouch(SeekBar seekBar) {}
+
+                    @Override
+                    public void onStopTrackingTouch(SeekBar seekBar) {
+                        // Automatically hide the popup after the user has selected a value.
+                        popupWindow.dismiss();
+                    }
+                });
+                popupWindow.showAsDropDown(v, 0, 0);
```

Comment:
This is a good place to update the state of the button, but see below my other comment about the background. Maybe we need to revert the checked/unchecked state, or use a different background drawable.

```java
mBinding.windowDistanceButton.setChecked(true);  // false?
popupWindow.setOnDismissListener(() -> mBinding.windowDistanceButton.setChecked(false));  // true?
popupWindow.showAsDropDown(v, 0, 0);
```

Reply:
Again, I am not sure the checked/unchecked status is the best for this scenario